### PR TITLE
Change default for attention mask creation

### PR DESCRIPTION
**Pull request by maanug-nv**
Since a fused attention kernel will be used in most cases (TE is required for Megatron Bridge), update `attention_mask_from_fusion` to avoid creating an attention mask in the batch that will not be used. Users can disable if needed. Applies to SFT only.

### CODE REVIEW

**Review by JRD971000 — APPROVED** [review]
lgtm
